air-quality-backend/src/air_quality/etl/forecast/forecast_adapter.py

```diff
@@ -107,3 +113,309 @@ def transform(forecast_data: ForecastData, locations: list[AirQualityLocation])
             }
             pollutant_forecast_for_location.append(document)
     return pollutant_forecast_for_location
+
+
+def _get_dimension_by_attr(
```

Comment:
Would it make sense to move these into an xarray_utils file or something?

- _get_dimension_by_attr
- _get_dim_names
- _normalise_data

We might find these functions useful elsewhere at some point, and they seem fairly generic (or could be, though I'm not sure how easy it would be to generify _get_dim_names)

Reply:
I moved _get_dimension_by_attr and get_dim_names to the new `shared.src.xarray_utils`. The normalise data is specific to the data textures. Did not find a good way for more generalisation, but maybe leave this for when we actually need it?